src/lib/optimizer/strategy/column_pruning_rule.cpp

```diff
@@ -242,6 +242,36 @@ void recursively_gather_required_expressions(
   }
 }
 
+void annotate_join_unused_inputs(
+    const std::shared_ptr<AbstractLQPNode>& node,
+    const std::unordered_map<std::shared_ptr<AbstractLQPNode>, ExpressionUnorderedSet>& required_expressions_by_node) {
```

Comment:
I am not deeply familiar with the column pruning rule. Does the additional responsibility of annotating unused inputs fit well with the original purpose of the rule?